File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        values = {}
        output = []

        frequency_buckets = [[] for i in range(len(nums) + 1)]

        for i in nums:

            if i in values.keys():
                values[i] += 1
            else:
                values[i] = 1
        
        for i in values.keys():
            print(f"key {i}")
            frequency_buckets[values[i] - 1].append(i)

        print(frequency_buckets)
        p = k - 1

        for i in range(len(nums)-1, -1, -1):
            for num in frequency_buckets[i]:
                output.append(num)

                if len(output) == k:
                    return output

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numbers = new HashSet<>();

        for (int number : nums) {
            numbers.add(number);
        }

        int longest = 0;
        for (int num : numbers) {
            if (!numbers.contains(num - 1)) {
                int currLongest = 1;
                int currNumber = num;

                while (numbers.contains(currNumber + 1)) {
                    currLongest++;
                    currNumber = currNumber + 1;
                }

                longest = Math.max(longest, currLongest);
            }
        }

        return longest;
    }
}
